component/src/stake/component.rs

```diff
@@ -635,16 +634,44 @@ pub(crate) trait StakingImpl: StateWriteExt {
 
         // Since we don't have a lookup from "addresses" to identity keys,
         // iterate over our app's validators, and match them up with the vote data.
-        for v in self.validator_list().await?.iter() {
-            let info = self
-                .validator_info(&v.identity_key)
-                .await?
-                .ok_or_else(|| anyhow::anyhow!("validator missing info"))?;
+        // We can fetch all the data required for processing each validator concurrently:
+        let mut js = JoinSet::new();
+        for v in self.validator_identity_list().await? {
```

Comment:
The real optimization here is only iterating over the list of validator _identity_ keys, rather than full validator data; the concurrent fetching is just additional gravy.